src/parallel.jl

```diff
@@ -223,6 +223,7 @@ function Base.put!(f!, ringbuffer::RingBuffer)
     return buf_
 end
 
+Base.put!(::Channel{T}, ::MLUtils.RingBuffer) = error("method not defined for RingBuffer")
```

Comment:
```suggestion
Base.put!(c::Channel{T}, b::MLUtils.RingBuffer) = throw(MethodError(put!, (c, b)))
```
Could we throw our own MethodError?